parquet/src/main/java/org/apache/iceberg/parquet/ParquetMetricsRowGroupFilter.java

```diff
@@ -423,4 +451,24 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
       return (T) conversions.get(id).apply(statistics.genericGetMax());
     }
   }
+
+  /**
+   * Checks against older versions of Parquet statistics which may have a null count but undefined min and max
+   * statistics. Returns true if nonNull values exist in the row group but no further statistics are available.
+   * <p>
+   * We can't use {@code  statistics.hasNonNullValue()} because it is inaccurate with older files and will return
+   * false if min and max are not set.
+   * <p>
+   * This is specifically for 1.5.0-CDH Parquet builds and later which contain the different unusual hasNonNull
+   * behavior. OSS Parquet builds are not effected because PARQUET-251 prohibits the reading of these statistics
```

Comment:
Why is this bug limited to binary and strings?

Reply:
Other types properly report min/max and hasNonNulls works. I'd have to go into the fix that CDH did to figure out for sure why their behavior is this way but I think they just disabled min/max stats to get around the Parquet-251 bug. 

Strings are stored as a binary type.

Reply:
Oh, I think I know. This probably happens when min/max values are so large that Parquet chooses not to keep them in stats, which isn't a problem for fixed-length types. That will happen in newer versions, too, but the stats object will be different.